framework/src/wifi_manager/wifi_manager.c

```diff
@@ -168,10 +183,13 @@ static void wifi_linkup_event_func(void)
 		nvdbg("CONNECTED FROM CLIENT - SOFT AP MODE");
 		wifi_status_set(CLIENT_CONNECTED);
 
+		wifi_mutex_release(w_mutex);
```

Comment:
here release called without accuire. is it right call?

Reply:
Please see line 151 where acquire is called.